src/home/OverviewLRTDistribution.jsx

```diff
@@ -2,67 +2,151 @@ import { Card, Progress } from '@nextui-org/react';
 import { Text } from '@visx/text';
-import { useMemo, useState } from 'react';
+import { useEffect } from 'react';
+import { useMutativeReducer } from 'use-mutative';
+import { useServices } from '../@services/ServiceContext';
+import { reduceState } from '../shared/helpers';
 import { formatNumber } from '../utils';
 
-const coins = [
-  { symbol: 'ADA', amount: 160, color: '#9C4FD9', inUSD: 122.48 },
-  { symbol: 'SOL', amount: 100, color: '#7BD94F', inUSD: 327.6 },
-  { symbol: 'BTA', amount: 80, color: '#AE7EDE', inUSD: 27363 },
-  { symbol: 'BTB', amount: 60, color: '#D94FB2', inUSD: 4363 },
-  { symbol: 'BTD', amount: 50, color: '#4FC0D9', inUSD: 1363 },
-  { symbol: 'BTE', amount: 40, color: '#D9B24F', inUSD: 363 },
-  { symbol: 'BTF', amount: 30, color: '#D9D34F', inUSD: 563 }
-];
-
-export const lrtData = {
-  wBETH: {
-    share: 2508659,
-    logo: '/wBETH.svg'
+const getChartColor = index => {
+  const colors = [
+    '#9C4FD9',
+    '#7BD94F',
+    '#AE7EDE',
+    '#D94FB2',
+    '#4FC0D9',
+    '#D9B24F',
+    '#D9D34F',
+    '#C9CBCF',
+    '#7BC225'
+  ];
+  return colors[index % colors.length];
+};
+
+const lrtMapping = {
+  bedrock: {
+    logo: '/bedrock.jpg'
+  },
+  eigenpie: {
+    logo: '/eigenpie.jpg'
+  },
+  etherfi: {
+    logo: '/etherfi.jpg'
   },
-  osETH: {
-    share: 8375748,
-    logo: '/osETH.svg'
+  euclid: {
+    logo: '/euclid.jpg'
   },
-  lsETH: {
-    share: 5141121,
-    logo: '/lsETH.svg'
+  inception: {
+    logo: '/inception.png'
   },
-  ankrETH: {
-    share: 6210246,
-    logo: '/ankrETH.svg'
+  kelp: {
+    logo: '/kelp.jpg'
   },
-  rETH: {
-    share: 6808644,
-    logo: '/rETH.svg'
+  prime: {
+    logo: '/prime.png'
   },
-  ETHx: {
-    share: 5373195,
-    logo: '/ETHx.svg'
+  puffer: {
+    logo: '/puffer.jpg'
   },
-  others: {
-    share: 9909455,
-    logo: '/image-group.svg'
+  swell: {
+    logo: '/swell.jpg'
   }
 };
 
 export default function OverviewLRTDistribution() {
-  const [active, setActive] = useState(null);
+  const { lrtService } = useServices();
+
+  const [state, dispatch] = useMutativeReducer(reduceState, {
+    activePieEntry: null,
+    isFetchingData: false,
+    lrtDistributionData: {}
+  });
   const width = 300;
   const half = width / 2;
 
-  const total = useMemo(
-    () =>
-      Math.floor(
-        coins.reduce((acc, coin) => acc + coin.amount * coin.inUSD, 0)
-      ),
-    []
-  );
+  const fetchLRTDistribution = async () => {
+    try {
+      dispatch({ isFetchingData: true });
+      const data = await lrtService.getLRTDistribution();
+      dispatch({ lrtDistributionData: data.protocols, isFetchingData: false });
+    } catch {
+      // TODO: Handle error
+      dispatch({
+        isFetchingTopOperators: false
+      });
+    }
+  };
 
-  const totalShare = Object.values(lrtData).reduce(
-    (sum, item) => sum + item.share,
-    0
-  );
+  useEffect(() => {
+    fetchLRTDistribution();
+  }, [lrtService, dispatch]);
```

Comment:
`nit`: The child dependencies are supposed to be specified in the `useCallback` instead. Or we move the function body in here and keep the same dep array
```suggestion
  }, [fetchLRTDistribution]);
```